scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -195,6 +206,43 @@ class FormatWriter(formatOps: FormatOps) {
     }
   }
 
+  private def checkOptionalBraces(locations: Array[FormatLocation]): Unit =
+    locations.foreach { floc =>
+      val ob = formatOps.OptionalBraces.get(floc.formatToken)(floc.style)
+      val ownerOpt = ob.flatMap(_.owner).filter {
+        /* if we add the end marker, it might turn a single-stat expression (or
```

Comment:
Let's mention it in the docs maybe so that people are not surprised if they set the new option to `1`

Reply:
done